packages/vue/src/vuerouter.ts

```diff
@@ -0,0 +1,66 @@
+import { captureException } from '@sentry/browser';
+import { Transaction, TransactionContext } from '@sentry/types';
+import VueRouter from 'vue-router';
+
+export type Action = 'PUSH' | 'REPLACE' | 'POP';
+
+export type Location = {
+  pathname: string;
+  action?: Action;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+} & Record<string, any>;
+
+export type VueRouterInstrumentation = <T extends Transaction>(
+  startTransaction: (context: TransactionContext) => T | undefined,
+  startTransactionOnPageLoad?: boolean,
+  startTransactionOnLocationChange?: boolean,
+) => void;
+
+let firstLoad = true;
+
+/**
+ * Creates routing instrumentation for Vue Router v2
+ *
+ * @param router The Vue Router instance that is used
+ */
+export function vueRouterInstrumentation(router: VueRouter): VueRouterInstrumentation {
+  return (
+    startTransaction: (context: TransactionContext) => Transaction | undefined,
+    startTransactionOnPageLoad: boolean = true,
+    startTransactionOnLocationChange: boolean = true,
+  ) => {
+    router.onError(error => captureException(error));
+
+    const tags = {
+      'routing.instrumentation': 'vue-router',
+    };
+    router.beforeEach((to, _from, next) => {
+      const data = {
+        params: to.params,
+        query: to.query,
+      };
+
+      if (startTransactionOnPageLoad && firstLoad) {
```

Comment:
Cannot `firstLoad` be detected by using `from === undefined` or something? This way we'll remove a global `firstLoad` state.

Reply:
`from` is never undefined and I am not sure if we always can rely on it comes from `/`.
This flag makes sure we are able to track pageload.

But I think I found another bug that I fixed by rewriting this section.